CHANGELOG.md

```diff
@@ -1,5 +1,13 @@
 ## unreleased
 
+## v4.14.0 - 2025.05.07
+
+* Update CSI driver for Kubernetes 1.33
```

Comment:
I think it's worth calling out changes in the csi manifests: i.e. that the (deprecated) validating webhook was removed, and that upstream csi components were updated to their latest versions.